
<!--==========================================================================
	FTL macros 
	========================================================================== -->


<!--==========================================================================
	CSS styles 
	========================================================================== -->


<style type="text/css">

body {
	background-color: #dddddd;
}

/* Table styles. */

table {
  border-color: #000;
  border-spacing: 0px;
  border-style: solid;
  border-width: 1px;
  cell-spacing: 0px;
}

.noBorder {
  border-width: 0px;
}

td, th {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 10pt;
  padding: 2px 0.5em;
  white-space: nowrap;
}

td.numeric {
  text-align: right;
}

th {
  background-color: #c0c0c0;
}

th.mainHeader {
  background-color: #808080;
  color: #ffffff;
  text-align: left;
}

th a {
  color: #000080;
  text-decoration: none;
}

th a:visited {
  color: #000080;
}

th a:active, th a:hover {
  color: #800000;
  text-decoration: underline;
}

.toc {
	border: 1px solid #aaa;
	background-color: #eeeeee;
	padding: 5px;
	font-size: 95%;
}

.note {
	border: 1px solid #aaa;
	background-color: #eeeeee;
	padding: 5px;
	font-size: 90%;
	text-align: left;
}

.main {
	margin-top: 15px;
	width: 1000px;
	text-align: left;
	background: #ffffff;
	border: 3px solid #777777;
}

.histo {
	table-layout:fixed;
	width:100%;
	border:1px solid #aaa;
	word-wrap:break-word;
}

</style>


<!--==========================================================================
	Index
	========================================================================== -->

<center>
<div class="main">

<center> <h3> SnpEff: Variant analysis </h3> </center>

<div style="margin-left: .5em">
<table class="toc"><tr><td>
	<center><b>Contents</b></center>
	<a href="#summary">Summary</a><br>
	<a href="#changeRateChr"> Variant rate by chromosome</a><br>
	<a href="#changesByType">Variants by type</a><br>
	<a href="#effectsImpact"> Number of variants by impact </a><br>
	<a href="#effectsImpact"> Number of variants by functional class </a><br>
	<a href="#effects"> Number of variants by effect </a><br>
	<a href="#quality">Quality histogram</a><br>
	<a href="#indels">InDel length histogram</a><br>
	<a href="#baseChages">Base variant table</a><br>
	<a href="#tstv">Transition vs transversions (ts/tv)</a><br>
	<a href="#alleleFreq"> Allele frequency </a><br>
	<a href="#alleleCount"> Allele Count </a><br>
	<a href="#codonChanges"> Codon change table </a><br>
	<a href="#aaChanges"> Amino acid change table </a><br>
	<a href="#chrChanges"> Chromosome variants plots </a><br>
	<a href="snpEff_genes.txt"> Details by gene </a><br>
</tr></td></table>
</div>


<!--==========================================================================
	Summary table 
	========================================================================== -->

<hr>
<a name="summary">
<center>
<b>Summary</b><p>

<table border=0>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> Genome </b> </td>
		<td> GRCh38.99 </td>
	</tr>
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Date </b> </td>
		<td> 2020-12-09 15:01 </td>
	</tr>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> SnpEff version </b> </td>
		<td> <pre>SnpEff 5.0 (build 2020-08-09 21:23), by Pablo Cingolani</pre> </td>
	</tr>
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Command line arguments </b> </td>
		<td> <pre>SnpEff  -o gatk GRCh38.99 variants/trio.filtered.vcf 
</pre> </td>
	</tr>
	<tr bgcolor=ffffff> 
		<td valign=top> <b> Warnings </b> </td>
		   
		<td bgcolor="#ffff00"> 73 </td>
	</tr>	
	<tr bgcolor=dddddd> 
		<td valign=top> <b> Errors </b> </td>
		 
		<td bgcolor="#dddddd"> 0 </td>
	</tr>	
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of lines (input file) </b> </td>
		<td> 556 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of variants (before filter) </b> </td>
		<td> 582 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of not variants <br>(i.e. reference equals alternative) </b> </td>
		<td> 0 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of variants processed <br> (i.e. after filter and non-variants) </b> </td>
		<td> 576 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of known variants <br>(i.e. non-empty ID) </b> </td>
		<td> 
		0
		( 0% ) 
		</td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Number of multi-allelic VCF entries <br>(i.e. more than two alleles) </b> </td>
		<td> 
		21
		</td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Number of effects </b> </td>
		<td> 10,357 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Genome total length </b> </td>
		<td> 63,147,197,748 </td>
	</tr>
	<tr bgcolor=dddddd>
		<td valign=top> <b> Genome effective length </b> </td>
		<td> 64,444,167 </td>
	</tr>
	<tr bgcolor=ffffff>
		<td valign=top> <b> Variant rate </b> </td>
		<td> 1 variant every 111,882 bases </td>
	</tr>
</table>
<p>
</center>

<!--==========================================================================
	Variant rate by chromosome
	========================================================================== -->

<hr> 
<a name="changeRateChr"> 

<center>
<b> Variants rate details </b><p>

<table border=1>
	<tr><th> Chromosome </th><th> Length </th><th> Variants </th><th> Variants rate </th></tr> 
		<tr>
			<td> 20 </td>
			<td class="numeric"> 64,444,167 </td>
			<td class="numeric"> 576 </td>
			<td class="numeric"> 111,882 </td>
		</tr> 
	<tr>
		<th> Total </th>
		<th class="numeric"> 64,444,167 </th>
		<th class="numeric"> 576 </th>
		<th class="numeric"> 111,882 </th>
	</tr> 
</table> 
</center>

<!--==========================================================================
	Variants by type
	========================================================================== -->

<hr> 
<a name="changesByType"> 
<center>
<b> Number variants by type</b><p>

<table border=1>
	<thead>
		<tr>
			<th> <b> Type   </b> </th>
			<th> <b> Total  </b> </th>
		</tr>
	</thead>
    <tbody>
	    <tr>
	    	<td> <b> SNP </b> </td>
	    	<td class="numeric" bgcolor="#ff0000"> 446 </td>
	    </tr>
	    <tr>
	    	<td> <b> MNP </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> INS </b> </td>
	    	<td class="numeric" bgcolor="#00ff00"> 65 </td>
	    </tr>
	    <tr>
	    	<td> <b> DEL </b> </td>
	    	<td class="numeric" bgcolor="#00ff00"> 65 </td>
	    </tr>
	    <tr>
	    	<td> <b> MIXED </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> INV </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> DUP </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> BND </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
	    <tr>
	    	<td> <b> INTERVAL </b> </td>
	    	<td class="numeric" bgcolor="ffffff"> 0 </td>
	    </tr>
    </tbody>
    <tfoot>
	    <tr>
	    	<th><b>Total </b> </th>
	    	<th class="numeric"> 576 </th>
	    </tr>
    </tfoot>
</table>
</center>

<!--==========================================================================
	Effects by impact
	========================================================================== -->

<hr> 
<a name="effectsImpact"> 
<center>
<b> Number of effects by impact </b> <p> 

<table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> LOW </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 5 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.048% </td>
		</tr>
		<tr> 
			<td> <b> MODERATE </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 2 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.019% </td>
		</tr>
		<tr> 
			<td> <b> MODIFIER </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 10,350 </td> 
			<td class="numeric" bgcolor="#ff0000"> 99.932% </td>
		</tr>
</table><br>
<p>
</center>

<!--==========================================================================
	Effects by functional class
	========================================================================== -->

<hr> 
<a name="effectsFuncClass"> 
<center>
<b> Number of effects by functional class </b> <p> 

<table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> MISSENSE </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 2 </td> 
			<td class="numeric" bgcolor="#00ff00"> 33.333% </td>
		</tr>
		<tr> 
			<td> <b> SILENT </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 4 </td> 
			<td class="numeric" bgcolor="#ff0000"> 66.667% </td>
		</tr>
</table><br>
<p>

Missense / Silent ratio: 0.5

</center>

<!--==========================================================================
	Effects stats
	========================================================================== -->

<hr> 
<a name="effects"> 
<center>
<b> Number of effects by type and region </b> <p> 

<table border=0>
	<tr>
		<th> Type </th>
		<th> Region </th>
	</tr>
	<tr>
		<td> <table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> DOWNSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#15e900"> 765 </td> 
			<td class="numeric" bgcolor="#15e900"> 7.386% </td>
		</tr>
		<tr> 
			<td> <b> EXON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#07f700"> 255 </td> 
			<td class="numeric" bgcolor="#07f700"> 2.462% </td>
		</tr>
		<tr> 
			<td> <b> INTRON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 9,021 </td> 
			<td class="numeric" bgcolor="#ff0000"> 87.101% </td>
		</tr>
		<tr> 
			<td> <b> NON_SYNONYMOUS_CODING </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 2 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.019% </td>
		</tr>
		<tr> 
			<td> <b> START_GAINED </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 1 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.01% </td>
		</tr>
		<tr> 
			<td> <b> SYNONYMOUS_CODING </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 4 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.039% </td>
		</tr>
		<tr> 
			<td> <b> UPSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#08f600"> 294 </td> 
			<td class="numeric" bgcolor="#08f600"> 2.839% </td>
		</tr>
		<tr> 
			<td> <b> UTR_3_PRIME </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 12 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.116% </td>
		</tr>
		<tr> 
			<td> <b> UTR_5_PRIME </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 3 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.029% </td>
		</tr>
</table><br>
 </td>
		<td> <table border=0>
	<thead>
		<tr> 
			<th><b> Type (alphabetical order)  </b></th>
			<th> &nbsp; </th>
			<th> Count </th>
			<th> Percent </th>
		</tr>
	</thead>
		<tr> 
			<td> <b> DOWNSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#15e900"> 765 </td> 
			<td class="numeric" bgcolor="#15e900"> 7.386% </td>
		</tr>
		<tr> 
			<td> <b> EXON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#07f700"> 261 </td> 
			<td class="numeric" bgcolor="#07f700"> 2.52% </td>
		</tr>
		<tr> 
			<td> <b> INTRON </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#ff0000"> 9,021 </td> 
			<td class="numeric" bgcolor="#ff0000"> 87.101% </td>
		</tr>
		<tr> 
			<td> <b> UPSTREAM </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#08f600"> 294 </td> 
			<td class="numeric" bgcolor="#08f600"> 2.839% </td>
		</tr>
		<tr> 
			<td> <b> UTR_3_PRIME </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00fe00"> 12 </td> 
			<td class="numeric" bgcolor="#00fe00"> 0.116% </td>
		</tr>
		<tr> 
			<td> <b> UTR_5_PRIME </b> </td> 
			<th> &nbsp; </th>
			<td class="numeric" bgcolor="#00ff00"> 4 </td> 
			<td class="numeric" bgcolor="#00ff00"> 0.039% </td>
		</tr>
</table><br>
 </td>
	</tr>
</table>
<p>

<img src="http://chart.apis.google.com/chart?cht=bvs&chbh=50,4,8&chs=800x300&chxt=y,y,x,x&chtt=Variations&chxp=1,50.0|3,50.0&chxr=0,0.0,87.10051173119629|1,0.0,100.0|3,0.0,100.0&chco=00FFFF,FA8072,FF0000,A52A2A,FFA500,DEB887,E9967A,7FFFD4,808080&chdl=Upstream|5%27UTR|Exon|Splice+Donor|Intron|Splice+Acceptor|3%27UTR|Downstream|Intergenic&chxl=1:|%25|2:|Intergenic|Up|5%27UTR|Exon|Donor|Intron|Acceptor|Exon|3%27UTR|Down|Intergenic|3:|&chd=e:AACFAAAAAAAAAAAAAAAAAA,AAAAACAAAAAAAAAAAAAAAA,AAAAAAB2AAAAAAB2AAAAAA,AAAAAAAAAAAAAAAAAAAAAA,AAAAAAAAAA..AAAAAAAAAA,AAAAAAAAAAAAAAAAAAAAAA,AAAAAAAAAAAAAAAAAFAAAA,AAAAAAAAAAAAAAAAAAFbAA,AAAAAAAAAAAAAAAAAAAAAA" border=1><p>

</center>

<!--==========================================================================
	Quality
	========================================================================== -->

<hr> 
<a name="quality"> 
<b> Quality:</b> 
<p>
<pre>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>39</td> </tr>
		<tr> <th>Max</th><td>4,066</td> </tr>
		<tr> <th>Mean</th><td>1,446.728</td> </tr>
		<tr> <th>Median</th><td>1,343</td> </tr>
		<tr> <th>Standard deviation</th><td>955.488</td> </tr>
		<tr> <th>Values</th><td>39,44,47,52,55,59,70,80,85,93,106,125,134,138,148,152,158,163,169,171,174,181,196,201,205,206,215,220,225,234,236,237,238,243,265,273,279,292,297,298,299,303,304,307,316,317,319,321,336,340,343,349,360,361,362,364,366,367,370,371,375,379,386,388,389,391,407,408,410,415,419,422,423,425,428,432,433,434,435,439,443,446,448,453,459,460,463,465,466,469,472,473,474,476,477,484,487,500,502,505,511,512,514,520,524,532,535,536,538,540,542,543,549,554,558,564,570,572,577,578,581,596,598,601,603,605,607,609,612,613,620,621,622,624,629,640,647,657,661,664,669,692,709,710,712,716,720,725,734,735,737,760,761,770,776,790,793,801,803,804,834,836,840,843,857,860,861,882,883,885,887,890,897,904,910,919,924,934,938,942,952,968,977,980,987,1002,1006,1013,1019,1020,1022,1023,1025,1038,1053,1055,1061,1062,1066,1067,1076,1082,1101,1102,1104,1105,1107,1108,1111,1116,1117,1122,1126,1132,1137,1144,1147,1157,1171,1172,1195,1198,1213,1238,1247,1279,1280,1289,1295,1298,1299,1312,1316,1325,1329,1330,1335,1336,1338,1340,1342,1344,1351,1369,1383,1390,1399,1402,1412,1424,1429,1431,1434,1440,1445,1454,1456,1463,1475,1487,1499,1501,1503,1505,1506,1513,1514,1518,1527,1529,1543,1558,1561,1566,1567,1569,1571,1578,1587,1589,1598,1600,1615,1636,1646,1657,1667,1676,1683,1691,1693,1696,1700,1715,1716,1720,1721,1724,1738,1747,1750,1757,1791,1792,1793,1801,1803,1814,1819,1824,1829,1843,1844,1855,1863,1872,1882,1885,1889,1893,1894,1898,1901,1903,1905,1911,1923,1926,1936,1941,1945,1950,1956,1960,1973,1974,1983,1990,1993,1995,1998,2001,2006,2011,2016,2017,2018,2039,2040,2042,2047,2050,2052,2058,2064,2068,2071,2073,2080,2086,2087,2095,2101,2117,2124,2137,2139,2143,2144,2162,2173,2184,2200,2211,2212,2214,2219,2222,2234,2241,2262,2271,2275,2283,2292,2302,2307,2349,2352,2358,2402,2407,2417,2429,2434,2444,2458,2465,2467,2472,2504,2513,2514,2518,2519,2521,2525,2540,2550,2557,2561,2568,2576,2589,2609,2620,2634,2638,2641,2651,2666,2674,2679,2680,2688,2690,2695,2720,2721,2727,2729,2733,2749,2751,2768,2771,2772,2777,2781,2782,2789,2800,2804,2826,2834,2836,2842,2881,2887,2921,2927,2935,2949,2959,2970,2994,3061,3095,3112,3122,3140,3167,3168,3171,3189,3205,3224,3279,3291,3300,3308,3310,3320,3336,3382,3408,3415,3434,3461,3465,3481,3546,3680,3741,3778,3834,3835,3908,4003,4066</td> </tr>
		<tr> <th>Count</th><td>1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,1,2,2,1,1,1,1,1,1,1,2,1,1,2,2,1,2,2,1,1,1,2,1,1,1,1,1,1,1,2,1,1,2,1,1,1,1,1,1,1,2,1,2,1,2,1,1,1,1,1,2,2,1,1,1,1,1,1,1,2,1,1,1,1,1,1,2,1,1,1,2,1,1,1,1,1,2,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,2,2,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,2,1,1,2,1,1,2,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,2,1,1,1,2,2,1,1,1,2,1,1,2,2,1,1,1,1,1,1,1,1,1,1,1,2,1,1,3,1,1,2,1,1,1,1,1,1,2,1,1,2,1,1,1,1,2,1,1,1,1,2,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,3,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,2,1,1,1,1,1,1,3,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,1,2,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,1,2,1,1,1,2,1,1,1,1,1,1,1,1,1,1,2,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1</td> </tr>
		</tr>
	</table>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Quality+histogram&chxr=0,1.0,4.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Count|2:|41||||||||||||||||||||||||||||||341||||||||||||||||||||||||||||||503||||||||||||||||||||||||||||||729||||||||||||||||||||||||||||||1061||||||||||||||||||||||||||||||1394||||||||||||||||||||||||||||||1718||||||||||||||||||||||||||||||2013||||||||||||||||||||||||||||||2423||||||||||||||||||||||||||||||2802|||||||||||||||||||||||||||||4066|3:|Quality&chd=e:VHVHAAVHVHAAVHVHAAVHAAVHqPVHqPqPAAVHVHAAqPAA..qPVHVHqPAAVHAAVHqPAAqPVHAAVHVHVHqPAAqPVHAAqPqPAAVHVHAAqPAAVHVHVHVHqPAAVHAAqPVHVHVHVHAAVHVHAAVHAAVHqPAA..VHAAqPVHAAVHAAVHVHAAqPqPAAVHAAVHVHAAVHVHAAVHqPAAVHAAVHVHVHVHVHAAVHVHAAqPAAVHVHAAVHqPAAqPAAqPVHAAVHVHAAqPVHAAVHAAVHVHAAVHVHAAVHVHAAqPAAVHVHAAqPVHAA..AAVHqPAA..VHAAVHVHAAVHAAqP..AAqPVHAAVHqPAAqPAAVHqPAAVHqPAAVHVHVHVHAAVHVHAAVHVHAAVHAAVHVHAAVHVHAAVHVHAAVHAAVHVHAAVH..AAqPAAVHVHAAVHVHAAVHVHAAVHAAVHVHAAqPVHAAqPVHAAVHAA..VHAAVHVHAAVHAAVHVHAAVHVHAAVHVHAAVHAAVHVHAAVHqPAAVHVHAAVHAAVHqPAAqPVHVHVHAAVHVHAAVHVHAAVHqPAAqPAAqPVHAAVHVHAAVHqPAAVHAAVHVHAAVHVHAAVHAA"><br>
</pre>

<!--==========================================================================
	InDels
	========================================================================== -->

<hr> 
<a name="indels"> 
<b> Insertions and deletions length:</b> 
<p>
<pre>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>0</td> </tr>
		<tr> <th>Max</th><td>26</td> </tr>
		<tr> <th>Mean</th><td>2.192</td> </tr>
		<tr> <th>Median</th><td>1</td> </tr>
		<tr> <th>Standard deviation</th><td>3.924</td> </tr>
		<tr> <th>Values</th><td>0,1,2,3,4,5,6,7,8,10,11,12,14,18,21,26</td> </tr>
		<tr> <th>Count</th><td>22,82,2,7,1,2,1,2,4,1,1,1,1,1,1,1</td> </tr>
		</tr>
	</table>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chg=2.0,8.0,3,2&chxp=1,50.0|3,50.0&chd=e:QA..ApEfAAApAAApB7AAAAAAAAAAAAAA&chxr=0,1.0,82.0|1,0.0,100.0|3,0.0,100.0&chtt=Insertion+deletion+length+histogram&chxl=1:|Count|2:|0||||||||||11|||||26|3:|Length"><br>
</pre>

<!--==========================================================================
	Base changes
	========================================================================== -->

<hr> 
<a name="baseChages"> 
<center>
<b> Base changes (SNPs) </b> <p>

<table border=1>
    <tr>
    	<td> &nbsp; </td>  <th> <b> A </b> </th> <th> <b> C </b> </th> <th> <b> G </b> </th> <th> <b> T </b> </th>
    </tr>
	    <tr> <th> <b> A </b> </th><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="#21dd00"> 23 </td><td class="numeric" bgcolor="#ff0000"> 82 </td><td class="numeric" bgcolor="#00ff00"> 14 </td> </tr>
	    <tr> <th> <b> C </b> </th><td class="numeric" bgcolor="#1ee100"> 22 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="#0ff000"> 18 </td><td class="numeric" bgcolor="#ff0000"> 82 </td> </tr>
	    <tr> <th> <b> G </b> </th><td class="numeric" bgcolor="#b04e00"> 61 </td><td class="numeric" bgcolor="#16e800"> 20 </td><td class="numeric" bgcolor="ffffff"> 0 </td><td class="numeric" bgcolor="#16e800"> 20 </td> </tr>
	    <tr> <th> <b> T </b> </th><td class="numeric" bgcolor="#1ae400"> 21 </td><td class="numeric" bgcolor="#b04e00"> 61 </td><td class="numeric" bgcolor="#1ee100"> 22 </td><td class="numeric" bgcolor="ffffff"> 0 </td> </tr>
</table>
</center>

<p>

<!--==========================================================================
	Ts/Tv
	========================================================================== -->

<hr> 
<a name="tstv">
<center> <b> Ts/Tv (transitions / transversions) </b> <p> </center>

<small>
<b>Note:</b> Only SNPs are used for this statistic.<br>
<b>Note:</b> This Ts/Tv ratio is a 'raw' ratio (ratio of observed events).
</small>
<p>

<center>
<table border=1>
	<tr> <th> Transitions </th><td class="numeric"> 921 </td> </tr>
	<tr> <th> Transversions </th><td class="numeric"> 508 </td> </tr>
	<tr> <th> Ts/Tv ratio </th><td class="numeric"> 1.813 </td> </tr>
</table>
</center>
<p>

	<b>All variants:</b>
	<pre>Sample ,father,mother,son,Total
Transitions ,323,264,334,921
Transversions ,172,145,191,508
Ts/Tv ,1.878,1.821,1.749,1.813
</pre>
	<p>
	<b>Only known variants</b> (i.e. the ones having a non-empty ID field):
	<pre>No results available (empty input?)</pre>
<p>

<!--==========================================================================
	Allele frequency
	========================================================================== -->

<hr> 
<a name="alleleFreq">
<center> <b> Allele frequency </b> <p> </center>

	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chg=8.0,13.0,3,2&chxp=1,50.0|3,50.0&chd=e:0eh6xRkeAA..Lh&chtt=Allele+Frequency+%25&chxr=0,2.0,141.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Count|2:|16||||||100|3:|Alleles+frequency"><br>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>16</td> </tr>
		<tr> <th>Max</th><td>100</td> </tr>
		<tr> <th>Mean</th><td>53.858</td> </tr>
		<tr> <th>Median</th><td>50</td> </tr>
		<tr> <th>Standard deviation</th><td>26.669</td> </tr>
		<tr> <th>Values</th><td>16,33,50,66,75,83,100</td> </tr>
		<tr> <th>Count</th><td>117,76,110,82,2,141,28</td> </tr>
		</tr>
	</table>
<p>
	
<!--==========================================================================
	Allele Count
	========================================================================== -->

<hr> 
<a name="alleleCount">
<center> <b> Allele Count </b> <p> </center>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chtt=Allele+Count&chxp=1,50.0|3,50.0&chd=e:zMeutbgA..AA&chxl=1:|Count|2:|1|||||6|3:|Allele+Count&chxr=0,25.0,141.0|1,0.0,100.0|3,0.0,100.0"><br>
	<table class="histo">
		<tr> <th width=15%>Min</th><td>1</td> </tr>
		<tr> <th>Max</th><td>6</td> </tr>
		<tr> <th>Mean</th><td>3.221</td> </tr>
		<tr> <th>Median</th><td>3</td> </tr>
		<tr> <th>Standard deviation</th><td>1.587</td> </tr>
		<tr> <th>Values</th><td>1,2,3,4,5,6</td> </tr>
		<tr> <th>Count</th><td>118,81,108,83,141,25</td> </tr>
		</tr>
	</table>
<p>


<!--==========================================================================
	Hom/Het table
	========================================================================== -->

<hr> 
<a name="homHet">
<center> <b> Hom/Het per sample </b> <p> </center>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chd=e:..z1AA&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Heterozygous+genotypes&chxr=0,186.0,319.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Heterozygous+count|2:|0||2|3:|Sample+number"><br>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chd=e:PXAA..&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Homozygous+%28ALT%29+genotypes&chxr=0,116.0,235.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Homozygous+count|2:|0||2|3:|Sample+number"><br>
	<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chd=e:......&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chtt=Missing+genotypes&chxr=0,0.0,4.0|1,0.0,100.0|3,0.0,100.0&chxl=1:|Missing+count|2:|0||2|3:|Sample+number"><br>
	
	<pre>Sample_names , father, mother, son
Reference , 88, 142, 131
Het , 319, 294, 186
Hom , 145, 116, 235
Missing , 4, 4, 4
</pre>
<p>

<!--==========================================================================
	Codon change table
	========================================================================== -->
	
<hr> 
<a name="codonChanges">
<center>
<b> Codon changes</b> <p>

	<div class="note">
		How to read this table: <br>
		- Rows are reference codons and columns are changed codons. E.g. Row 'AAA' column 'TAA' indicates how many 'AAA' codons have been replaced by 'TAA' codons.<br>
		- Red background colors indicate that more changes happened (heat-map).<br>
		- Diagonals are indicated using grey background color <br> 
		- WARNING: This table may include different translation codon tables (e.g. mamalian DNA and mitochondrial DNA).<br>
		<p>
	</div><p>

  
<table border=1>
	<thead>
		<tr> 
			<th> &nbsp; </th>
			 <th> CAG </th>  <th> CTG </th>  <th> GCG </th>  <th> GCT </th>  <th> GGA </th>  <th> GGG </th> 
		</tr>
	</thead>
			<tr> 
				<th> CAG </th>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> CTG </th>
					
						<td class="numeric" bgcolor="ffffff"> 2 </td>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> GCG </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> GCT </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> 2 </td>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> GGA </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> 2 </td>
			</tr>
			<tr> 
				<th> GGG </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> &nbsp </th>
			</tr>
</table>
</center>

<!--==========================================================================
	Amino acid change table
	========================================================================== -->

<hr> 
<a name="aaChanges">
<center>
<b> Amino acid changes</b><p> 

	<div class="note">
		How to read this table: <br>
		- Rows are reference amino acids and columns are changed amino acids. E.g. Row 'A' column 'E' indicates how many 'A' amino acids have been replaced by 'E' amino acids.<br>
		- Red background colors indicate that more changes happened (heat-map).<br>
		- Diagonals are indicated using grey background color <br> 
		- WARNING: This table may include different translation codon tables (e.g. mamalian DNA and mitochondrial DNA).<br>
		<p>
	</div><p>

<table border=1>
	<thead>
		<tr> 
			<th> &nbsp; </th>
			 <th> A </th>  <th> G </th>  <th> L </th>  <th> Q </th> 
		</tr>
	</thead>
			<tr> 
				<th> A </th>
					
						<th class="numeric"> 2 </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> G </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> 2 </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
			</tr>
			<tr> 
				<th> L </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> &nbsp </th>
					
						<td class="numeric" bgcolor="ffffff"> 2 </td>
			</tr>
			<tr> 
				<th> Q </th>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<td class="numeric" bgcolor="ffffff"> &nbsp </td>
					
						<th class="numeric"> &nbsp </th>
			</tr>
</table>
</center>

<!--==========================================================================
	Chromosome variants table
	========================================================================== -->

<hr> 
<a name="chrChanges">
<center>
<b> Variants by chromosome</b><p> 

<center>
	<pre>
		<img src="http://chart.apis.google.com/chart?cht=bvg&chbh=8,4,2&chs=800x300&chxt=y,y,x,x&chxp=1,50.0|3,50.0&chg=6400000.0,57.0,3,2&chtt=Variants+histogram%3A+20&chxr=0,0.0,576.0|1,0.0,100.0|3,0.0,100.0&chd=e:AAAAAAAAAAAAAAAAAAAA..AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&chxl=1:|Variants%2F1000Kb|2:|0||||||||||10000000||||||||||20000000||||||||||30000000||||||||||40000000||||||||||50000000||||||||||60000000||||64000000|3:|Position"><br>
		20, Position,0,1000000,2000000,3000000,4000000,5000000,6000000,7000000,8000000,9000000,10000000,11000000,12000000,13000000,14000000,15000000,16000000,17000000,18000000,19000000,20000000,21000000,22000000,23000000,24000000,25000000,26000000,27000000,28000000,29000000,30000000,31000000,32000000,33000000,34000000,35000000,36000000,37000000,38000000,39000000,40000000,41000000,42000000,43000000,44000000,45000000,46000000,47000000,48000000,49000000,50000000,51000000,52000000,53000000,54000000,55000000,56000000,57000000,58000000,59000000,60000000,61000000,62000000,63000000,64000000
20,Count,0,0,0,0,0,0,0,0,0,0,576,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0

	</pre>
</center>

<!--==========================================================================
	Variants by gene 
	========================================================================== -->

<center>
	<hr>
	<b> Details by gene </b><p>
	<p>
	<b><a href="snpEff_genes.txt">Here</a></b> you can find a tab-separated table.
</center>
</div>
</center>